components/content-service/pkg/storage/s3.go

```diff
@@ -289,33 +290,36 @@ func (s3st *s3Storage) DownloadSnapshot(ctx context.Context, destination string,
 	return s3st.download(ctx, destination, name, mappings)
 }
 
+// download object using s5cmd (prior to which we used aws sdk)
 func (s3st *s3Storage) download(ctx context.Context, destination string, obj string, mappings []archive.IDMapping) (found bool, err error) {
-	downloader := s3manager.NewDownloader(s3st.client, func(d *s3manager.Downloader) {
-		d.Concurrency = defaultCopyConcurrency
-		d.PartSize = defaultPartSize * megabytes
-		d.BufferProvider = s3manager.NewPooledBufferedWriterReadFromProvider(25 * megabytes)
-	})
-
-	s3File, err := os.CreateTemp("", "temporal-s3-file")
+	tempFile, err := os.CreateTemp("", "temporal-s3-file")
 	if err != nil {
 		return true, xerrors.Errorf("creating temporal file: %s", err.Error())
 	}
-	defer os.Remove(s3File.Name())
+	tempFile.Close()
 
-	_, err = downloader.Download(ctx, s3File, &s3.GetObjectInput{
-		Bucket: aws.String(s3st.Config.Bucket),
-		Key:    aws.String(obj),
-	})
+	args := []string{
+		"--numworkers", "20",
+		"cp", "--part-size", "25",
```

Comment:
Why adjust part-size?

Reply:
Reducing from 50 (default) to reliably removed .3s from each test run. I'll test again with the default though, when using `--concurrency`. Thank you for the catch, I'll do a test tomorrow, and then deploy to `dev-internal` to test.

Reply:
Attached a gist to the PR with test results, this actually helped still.